libs/vkd3d/vkd3d_private.h

```diff
@@ -403,12 +403,88 @@ static inline void vkd3d_private_store_destroy(struct vkd3d_private_store *store
     pthread_mutex_destroy(&store->mutex);
 }
 
+static inline HRESULT vkd3d_private_data_lock(struct vkd3d_private_store *store)
+{
+    int rc;
+    if ((rc = pthread_mutex_lock(&store->mutex)))
+    {
+        ERR("Failed to lock mutex, error %d.\n", rc);
+        return hresult_from_errno(rc);
+    }
+
+    return S_OK;
+}
+
+static inline void vkd3d_private_data_unlock(struct vkd3d_private_store *store)
+{
+    pthread_mutex_unlock(&store->mutex);
+}
+
 HRESULT vkd3d_get_private_data(struct vkd3d_private_store *store,
         const GUID *tag, unsigned int *out_size, void *out);
-HRESULT vkd3d_set_private_data(struct vkd3d_private_store *store,
-        const GUID *tag, unsigned int data_size, const void *data);
-HRESULT vkd3d_set_private_data_interface(struct vkd3d_private_store *store,
-        const GUID *tag, const IUnknown *object);
+
+HRESULT vkd3d_private_store_set_private_data(struct vkd3d_private_store *store,
+        const GUID *tag, const void *data, unsigned int data_size, bool is_object);
+
+static inline HRESULT vkd3d_set_private_data(struct vkd3d_private_store *store,
+        const GUID *tag, unsigned int data_size, const void *data)
+{
+    return vkd3d_private_store_set_private_data(store, tag, data, data_size, false);
+}
+
+static inline HRESULT vkd3d_set_private_data_interface_locked(struct vkd3d_private_store *store,
+        const GUID *tag, const IUnknown *object)
+{
+    const void *data = object ? object : (void *)&object;
+    HRESULT hr;
+
+    if (FAILED(hr = vkd3d_private_data_lock(store)))
+        return hr;
+
+    if (FAILED(hr = vkd3d_private_store_set_private_data(store, tag, data, sizeof(object), !!object)))
+    {
+        vkd3d_private_data_unlock(store);
+        return hr;
+    }
+
+    vkd3d_private_data_unlock(store);
+    return hr;
+}
+
+static inline bool vkd3d_private_data_object_name_ptr(REFGUID guid,
+    UINT data_size, const void *data, const char **out_name)
+{
+    *out_name = NULL;
+
+    if (IsEqualGUID(guid, &WKPDID_D3DDebugObjectName))
+    {
+        const char *name = (const char *)data;
+
+        if (!data || data_size < sizeof(char))
```

Comment:
Just check data_size == 0, sizeof(char) is defined to be 1.

Reply:
Was keeping it consistent with the WCHAR path below, but can do